master/build-rust-manifest.py

```diff
@@ -0,0 +1,489 @@
+import os
+import sys
+import urllib2
+import shutil
+import subprocess
+import re
+import hashlib
+
+# The release channel name
+channel = sys.argv[1]
+
+# The archive date for the final 'rust' release artifacts,
+# as determined by rust-buildbot
+today = sys.argv[2]
+
+# The s3 http address to use for querying info about packages.  Use
+# this instead of public_addy because it's not behind cloudfront,
+# which sometimes disagrees with the s3 bucket.
+#
+# Among other things, this addressed is used to retrieve the hashes
+# of the components, which are then embedded in the manifest.
+#
+# FIXME: There is a security problem here, with using http to build
+# the manifests. It may be lessened by the build master running on
+# aws. So amazon is routing our traffic, and they are not going to
+# mitm us.
+s3_addy = sys.argv[3]
+
+# The live address at which packages will be downloaded,
+# i.e. https://static.rust-lang.org. This is the public
+# version of s3_addy, for encoding in manifests.
+public_addy = sys.argv[4]
+
+# The directory containing the 'rust' packages as output
+# by rust-packaging. These have been previously generated
+# by rust-buildbot, but not yet uploaded to s3
+rust_package_dir = sys.argv[5]
+
+# The file to write the manifest to
+out = sys.argv[6]
+
+print "channel: " + channel
+print "today: " + today
+print "s3_addy: " + s3_addy
+print "public_addy: " + public_addy
+print "rust_package_dir: " + rust_package_dir
+print "out: " + out
+print
+
+# These are the platforms we produce compilers for
+host_list = sorted([
+    "x86_64-unknown-linux-gnu",
+    "i686-unknown-linux-gnu",
+    "x86_64-pc-windows-gnu",
+    "i686-pc-windows-gnu",
+    "x86_64-pc-windows-msvc",
+    "i686-pc-windows-msvc",
+    "x86_64-apple-darwin",
+    "i686-apple-darwin",
+])
+
+# These are the platforms we produce standard libraries for
+target_list = sorted([
+    "aarch64-unknown-linux-gnu",
+    "arm-linux-androideabi",
+    "arm-unknown-linux-gnueabif",
+    "arm-unknown-linux-gnueabihf",
+    "i686-apple-darwin",
+    "i686-pc-windows-gnu",
+    "i686-pc-windows-msvc",
+    "i686-unknown-linux-gnu",
+    "mips-unknown-linux",
+    "mipsel-unknown-linux",
+    "x86_64-apple-darwin",
+    "x86_64-pc-windows-gnu",
+    "x86_64-pc-windows-msvc",
+    "x86_64-unknown-linux-gnu",
+    "x86_64-unknown-linux-musl",
+])
```

Comment:
There's also a list of these kinda in master.cfg which we'll need to keep in sync, but probably too difficult for now to refactor